### PR TITLE
feat: Add PostgreSQL and layered architectur home work

### PR DESCRIPTION
**Pull request by SobollEva**

### CODE REVIEW

**Comment by SobollEva** [comment]
User_db-table_users
User_db-table_users_data

File: src/hw-3/models/user.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+export function UserModel(sequelize: any, DataTypes: any) {
```
Comment (pavel-rudzenia) [inline src/hw-3/models/user.ts]: I think it doesn't have much sense to create userModel each time you use it, better to make it singleton.

File: src/hw-3/models/user.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        password: {
+            type: DataTypes.TEXT,
+            allowNull: false
+        },
+        age: {
+            type: DataTypes.INTEGER,
+            allowNull: false
+        },
+        isDeleted: {
+            type: DataTypes.BOOLEAN
```
Comment (pavel-rudzenia) [inline src/hw-3/models/user.ts]: I'd add default value here, like `defaultValue: false`.

File: src/hw-3/routers/user.router.ts [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+            } else {
+                res.status(404).json({message: `User with id ${req.params.id} is not found`});
+            }
+        } catch (e){
+            res.status(404).json({message: ERROR_MESSAGE});
+        }
+    });
+
+router.route('/user')
+    .post(validation(), async (req: express.Request, res: express.Response) => {
```
Comment (pavel-rudzenia) [inline src/hw-3/routers/user.router.ts:57]: btw, you could also add validation layer to your db model

File: src/hw-3/services/user.sevice.ts [hunk 4]
```diff
@@ -0,0 +1,60 @@
+import { NewUser, User } from '../types/user';
+import { Sequelize, DataTypes, Op } from 'sequelize';
+import { UserModel } from '../models/user';
+
+const sq = new Sequelize('postgres://postgres:genya147@localhost:5433/user_db');
```
Comment (pavel-rudzenia) [inline src/hw-3/services/user.sevice.ts]: be honest I don't like that it's hardcoded. I'd like if you used `process.env` here to configure postgres username/password/host/port/db. As there are a lot of params, `dotenv` would work even better.

File: src/hw-3/services/user.sevice.ts [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+        return await UserModel(sq, DataTypes)
+            .update(
+                { isDeleted: true },
+                {
+                    where: { id }
+                })
+    }
+
+    async createUser(newUser: NewUser): Promise<User> {
+        const user: User = {...newUser, id: uuidv4(newUser), isDeleted: false};
```
Comment (pavel-rudzenia) [inline src/hw-3/services/user.sevice.ts]: I think you can use sequelize model/UUID data type to generate id of the user

File: src/hw-3/models/user.ts (hunk 1 above)
Comment (SobollEva) [reply src/hw-3/models/user.ts]: Fixed.

File: src/hw-3/models/user.ts (hunk 2 above)
Comment (SobollEva) [reply src/hw-3/models/user.ts]: Added.

File: src/hw-3/routers/user.router.ts (hunk 3 above)
Comment (SobollEva) [reply src/hw-3/routers/user.router.ts:57]: Which kind of validation do you mean? There're type and required fields validation in scheme (https://github.com/SobollEva/nodejs_ep_2022/blob/3bee3afc37f562c50dcec8769f873ec69e20b024/src/hw-3/utils/validation.utils.ts).

File: src/hw-3/services/user.sevice.ts (hunk 4 above)
Comment (SobollEva) [reply src/hw-3/services/user.sevice.ts]: Fixed. `dotenv` has been added, .env has been created.

File: src/hw-3/services/user.sevice.ts (hunk 5 above)
Comment (SobollEva) [reply src/hw-3/services/user.sevice.ts]: Added, thanks.

**Pull request merged**